src/main/java/com/meilisearch/sdk/json/GsonJsonHandler.java

```diff
@@ -45,8 +50,7 @@ public <T> T decode(Object o, Class<?> targetClass, Class<?>... parameters) thro
                 return gson.fromJson((String) o, parameterized.getType());
             }
         } catch (Exception e) {
-            // todo: use dedicated exception
-            throw new RuntimeException("Error while deserializing: ", e);
+            throw new MeiliSearchRuntimeException(e);
```

Comment:
I think we can create a new custom class of errors to describe the decoding exception here :)